src/main/java/com/jcabi/http/wire/AbstractHeaderBasedCachingWire.java

```diff
@@ -92,7 +92,9 @@ public final Response send(
         final InputStream content, final int connect, final int read
     ) throws IOException {
         final Response rsp;
-        if (method.equals(Request.GET)) {
+        if (method.equals(Request.GET)
+            && !this.requestHasCmcHeader(headers)
+        ) {
```

Comment:
@amihaiemil I think you can join these lines, `        if (method.equals(Request.GET) && !this.requestHasCmcHeader(headers))` has less < 80 chars